clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1314,13 +1332,36 @@ private void recordTopicFetchMetrics(String topic, int bytes, int records) {
         private void updatePartitionLagSensors(Set<TopicPartition> assignedPartitions) {
             if (this.assignedPartitions != null) {
                 for (TopicPartition tp : this.assignedPartitions) {
-                    if (!assignedPartitions.contains(tp))
+                    if (!assignedPartitions.contains(tp)) {
                         metrics.removeSensor(partitionLagMetricName(tp));
+                    }
                 }
             }
             this.assignedPartitions = assignedPartitions;
         }
 
+        private void updatePartitionLeadSensors() {
+            metrics.removeSensor(partitionLeadSensorName());
+            this.leadMetricAddedPartitions.clear();
+        }
+
+        private void recordPartitionLead(TopicPartition tp, long lead) {
+            this.recordsFetchLead.record(lead);
+
+            Sensor recordsLead = this.metrics.sensor(partitionLeadSensorName(), this.recordsFetchLead);
+            if (!this.leadMetricAddedPartitions.contains(tp)) {
+                this.leadMetricAddedPartitions.add(tp);
+                Map<String, String> tags = new HashMap<>(2);
+                tags.put("topic", tp.topic().replace('.', '_'));
+                tags.put("partition", String.valueOf(tp.partition()));
```

Comment:
String.valueOf(tp.partition()) can just be tp.partition().toString()?

Reply:
`partition` in `TopicPartition` is of `int` type instead of `Integer`, which is a primitive type, no `toString` method.